src/main/java/seedu/address/model/property/UniqueList.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.model.property;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.property.exceptions.DuplicateListableException;
+import seedu.address.model.property.exceptions.ListableNotFoundException;
+
+/**
+ * A abstract list that enforces uniqueness between its elements and does not allow nulls.
```

Comment:
`UniqueList` is not a abstract class.
```suggestion
 * A generic list that enforces uniqueness between its elements and does not allow nulls.
```

Reply:
I forgot to change it after I decided to use generics instead, thanks for highlighting